pkg/eniconfig/eniconfig.go

```diff
@@ -141,10 +142,9 @@ func GetNodeSpecificENIConfigName(ctx context.Context, k8sClient client.Client)
 	}
 
 	eniConfigName = val
-	if val != eniConfigDefault {
-		labels := node.GetLabels()
-		labels["vpc.amazonaws.com/eniConfig"] = eniConfigName
-		node.SetLabels(labels)
+	if val != eniConfigDefault && generate_event {
+		// Signal event to VPC RC that the custom networking is enabled
+			eventRecorder.SendNodeEvent(corev1.EventTypeNormal, "AwsNodeNotification", "CustomNetworkingEnabled", eniConfigLabel+"="+eniConfigName)
```

Comment:
Why generate the event from here though? Why not from the IPAMD init flow where we currently check the env variable and are currently labeling the node? - https://github.com/aws/amazon-vpc-cni-k8s/blob/master/pkg/ipamd/ipamd.go#L543. We can avoid the eventRecorder references in the controller code flow here...

Reply:
Thanks, Apurup. I have move the code to ipamd instead.